contracts/jobs/JobsManager.sol

```diff
@@ -136,21 +128,22 @@ contract JobsManager is IJobsManager, Verifiable, Manager {
      * @dev Deposit funds for jobs
      * @param _amount Amount to deposit
      */
-    function deposit(uint256 _amount) external whenSystemNotPaused returns (bool) {
+    function deposit(uint256 _amount) external afterInitialization whenSystemNotPaused returns (bool) {
         broadcasterDeposits[msg.sender] = broadcasterDeposits[msg.sender].add(_amount);
-        // Transfer tokens for deposit. Sender needs to approve amount first
-        token.transferFrom(msg.sender, this, _amount);
+        // Transfer tokens for deposit to Minter. Sender needs to approve amount first
+        livepeerToken().transferFrom(msg.sender, minter(), _amount);
 
         return true;
     }
 
     /*
      * @dev Withdraw deposited funds
+     * FIXME: Attacker can withdraw funds before transcoder has a chance to claim them
```

Comment:
Going to need logic to make sure this method is not used to withdraw funds before a transcoder has a chance to claim them. Will track with an issue